block/retriever.go

```diff
@@ -146,7 +143,8 @@ func (m *Manager) ProcessNextDABatch(daMetaData *da.DASubmitMetaData) error {
 				continue
 			}
 
-			err := m.applyBlock(block, batch.Commits[i], types.BlockMetaData{Source: types.DA, DAHeight: daMetaData.Height})
+			// We dont validate because validateBlockBeforeApply already checks if the block is already applied, and we don't need to fail there.
+			err := m.applyBlockWithFraudHandling(block, batch.Commits[i], types.BlockMetaData{Source: types.DA, DAHeight: daMetaData.Height}, false)
```

Comment:
or remove the validate param in the function since it will be always validating then...

Reply:
i think if it fails to validate here it should not continue and notify fraud, in the same way it is done when applying block from p2p. therefore imo it is not necessary to have the validateBlockBeforeApply  call before applyBlockWithFraudHandling, since it could be called  inside the applyBlockWithFraudHandling (same way as p2p). this way the validate param in  applyBlockWithFraudHandling function is not really necessary, because it wont be used.